doc: recommend not using libuv node-api function (#53521)
nodejs/node: doc/api/n-api.md

* `[in] env`: The environment that the API is invoked under.
* `[out] loop`: The current libuv loop instance.

> Note while libuv has been relatively stable over time, it does

Reviewer: ```suggestion
Note: While libuv has been relatively stable over time, it does
```

> not provide an ABI stability guarrantee. Avoid using this fuction.

Reviewer: ```suggestion
not provide an ABI stability guarrantee. Use of this function should be avoided.
```

> Its use may result in an addon that does not work across Node.js versions.

Reviewer 2: maybe suggest an alternative?

Reviewer 3: Yeah, it might be worth mentioning https://nodejs.org/docs/latest/api/n-api.html#asynchronous-thread-safe-function-calls

Author: added

* `[in] env`: The environment that the API is invoked under.
* `[out] loop`: The current libuv loop instance.

Note: While libuv has been relatively stable over time, it does

> not provide an ABI stability guarrantee. Use of this function should be avoided.

Reviewer 4: ```suggestion
not provide an ABI stability guarantee. Use of this function should be avoided.
```
